numba/core/datamodel/models.py

```diff
@@ -899,6 +899,13 @@ def __init__(self, dmm, fe_type):
         self._be_type = dmm.lookup(fe_type.dtype).get_data_type()
         super(NestedArrayModel, self).__init__(dmm, fe_type)
 
+    def as_storage_type(self):
+        """Return the LLVM type representation for the storage of
+        the nestedarray.
+        """
+        ret = ir.ArrayType(self._be_type, self._fe_type.nitems)
```

Comment:
NOTE: I think this is ok as the type system only allows nested arrays of simple numerical types.